src/Microsoft.Azure.SignalR.Common/ServiceConnectionBase.cs

```diff
@@ -322,6 +326,9 @@ private async Task ProcessIncomingAsync()
 
                             UpdateReceiveTimestamp();
 
+                            // No matter what kind of message come in, trigger send ping check
+                            await TrySendPingAsync();
```

Comment:
If bulk messages coming, _= will start many tasks at same time. Will this affect performance?

---
In reply to: [237727468](https://github.com/Azure/azure-signalr/pull/280#discussion_r237727468) [](ancestors = 237727468)

Reply:
why *start many tasks at the same time*?

---
In reply to: [237745336](https://github.com/Azure/azure-signalr/pull/280#discussion_r237745336) [](ancestors = 237745336,237727468)

Reply:
Every message will start a task to TrySendPingAsync(). But since read message is sequential and most of the time it only do check time in TrySendPingAsycn() which is fast, you're right they won't be together. Will update to _=.

---
In reply to: [238146901](https://github.com/Azure/azure-signalr/pull/280#discussion_r238146901) [](ancestors = 238146901,237745336,237727468)